src/org/sosy_lab/cpachecker/cfa/parser/llvm/LlvmTypeConverter.java

```diff
@@ -210,26 +214,26 @@ private CType getFunctionType(TypeRef pFuncType) {
     return new CFunctionType(returnType, parameterTypes, takesVarArgs);
   }
 
-  private CType getIntegerType(final int pIntegerWidth) {
-    return new CBitFieldType(CNumericTypes.INT, pIntegerWidth);
+  private CType getIntegerType(final int pIntegerWidth, final boolean isUnsigned) {
+    return new CBitFieldType(isUnsigned ? CNumericTypes.UNSIGNED_INT : CNumericTypes.INT, pIntegerWidth);
   }
 
-  private CType getFloatType(final TypeKind pType) {
+  private CType getFloatType(final TypeKind pType, final boolean isUnsigned) {
 
     switch (pType) {
       case Half:
         // FIXME: This is actually wrong, but at the time of this writing
         // we have no way of defining a float of 16bit width
-        return getSimplestCType(CBasicType.FLOAT);
+        return getSimplestCType(CBasicType.FLOAT, isUnsigned);
 
       case Float:
-        return getSimplestCType(CBasicType.FLOAT);
+        return getSimplestCType(CBasicType.FLOAT, isUnsigned);
 
       case Double:
         if (machineModel.getSizeofDouble() * 8 == 64) {
-          return getSimplestCType(CBasicType.DOUBLE);
+          return getSimplestCType(CBasicType.DOUBLE, isUnsigned);
         } else if (machineModel.getSizeofLongDouble() * 8 == 64) {
-          return getSimplestCType(CBasicType.DOUBLE, true);
+          return getSimplestCType(CBasicType.DOUBLE, isUnsigned, true /* is long */);
```

Comment:
It's very good practice to add comments to the boolean parameter,
and it's probably not important, but since we are using the ErrorProne-Compiler,
could you use [error prone's convention](https://github.com/google/error-prone/blob/master/docs/bugpattern/ParameterName.md) for the comments?
This way we'll stay consistent from the start.

Reply:
Sure.